### PR TITLE
[feat] Add filters on View Plant page

### LINKED ISSUE #28: Add Filters on View Plant Page
see https://linear.app/trap-garden/issue/TG-27/add-filters-on-view-plant-page

### PR DESCRIPTION
**Pull request by kylezryr**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Added filters for growing season, sunlight, and difficulty level on view plants page
- Similar functionality to `seasonal-planting-guide` page, results change depending on filter(s) selected

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
image
image
image

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Go to `/view-plants` page 
- Input some filters and check that what is displayed on both 'My Plants' and 'All Plants' are correct 
- Check that clear filters button works 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Search bar
- Styling

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: utils/helpers.ts [hunk 1]
```diff
@@ -151,3 +151,62 @@ export function checkSearchTerm(searchTerm: string, plant: Plant) {
...
+  ]);
+
+  // For each sunlight selected, check if plant's min_hours and max_hours are
+  // within that enum's range
+  const sunlightBoolean: boolean[] = [];
+  for (const sunlight of sunlightFilterValue) {
+    const [minHours, maxHours] = sunlightToHours.get(sunlight.value)!;
+    // if max_hours is null then plant can only receive min_hours of sunlight
+    if (plant.sunlight_max_hours === null) {
+      sunlightBoolean.push(
```
Comment (ccatherinetan) [inline utils/helpers.ts]: nit: instead of creating an array `sunlightBoolean` and pushing to it for each sunlight value, we can just directly return true if the plant matches any of the conditions

`if (plant.sunlight_min_hours >= minHours && plant.sunlight_min_hours <= maxHours) return true;`

and `return false` at the end. 

I think the current implementation is fine, but I'm just curious if there was a reason (i.e. if there was any benefit) over the alternate implementation suggested here.

File: utils/helpers.ts [hunk 2]
```diff
@@ -151,3 +151,62 @@ export function checkSearchTerm(searchTerm: string, plant: Plant) {
...
+  plant: Plant,
+) {
+  // Automatically returns true if no selected difficulty
+  if (difficultyFilterValue.length === 0) {
+    return true;
+  }
+
+  // For each difficulty selected,
+  // check if plant's difficulty_level matches difficulty
+  const difficultyBoolean: boolean[] = [];
```
Comment (ccatherinetan) [inline utils/helpers.ts]: same comment about the arrays - just noting it here
Honestly, i think it's ok to leave it as it is now, since the logic is correct, but I just wanted to flag this in case we wanted to change it in the future.

**Review by ccatherinetan — APPROVED** [review]
looking great; tysm kyle! i think it's ready to merge! next, we'll need to style the search bar and filter buttons. if it's hard to style the filter buttons from the library, we might need to make a custom multi select button.

File: app/view-plants/page.tsx [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
-import { useEffect, useState } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import { UUID } from 'crypto';
 import supabase from '@/api/supabase/createClient';
 import { getAllPlants, getPlantById } from '@/api/supabase/queries/plants';
+import FilterDropdownMultiple from '@/components/FilterDropdownMultiple';
 import PlantCard from '@/components/PlantCard';
-import { Plant } from '@/types/schema';
+import SearchBar from '@/components/SearchBar';
+import { DropdownOption, Plant } from '@/types/schema';
```
Comment (ccatherinetan) [inline app/view-plants/page.tsx:10]: I'm thinking of having Kevin refactor the DropdownOption type into DropdownOption<T> defined below (i.e. making DropdownOption a generic)
```
interface DropdownOption<T> {
   label: string; 
   value: T
}
```
~Can you also include that change in schema.d.ts and make sure that all the typing of `DropdownOption[]` is replaced with `DropdownOption<string>[]`?~

EDIT: jk, i think he's defined it as `DropdownOption<T = string>`, in which case we should be chilling
